consensus/scp/tests/mock_network/mod.rs

```diff
@@ -414,41 +328,63 @@ impl SCPNode {
                                 SCPNodeTaskMessage::StopTrigger => {
                                     break 'main_loop;
                                 }
-                            };
+                            },
+                            Err(_) => {
+                                // Yield to other threads when we don't get a new message
+                                // This improves performance significantly.
+                                std::thread::yield_now();
+                            }
+                        };
+
+                        let incoming_msgs_count = incoming_msgs.len();
+
+                        if !(incoming_msgs_count == 0 || incoming_msgs_count == 1) {
+                            log::error!(logger, "incoming_msgs_count > 1");
+                            assert!(incoming_msgs_count == 0 || incoming_msgs_count == 1);
+                            // exit
                         }
 
                         // Process values submitted to our node
                         if !pending_values.is_empty() {
-                            let vals = pending_values
-                                              .iter()
-                                              .cloned()
-                                              .collect::<Vec<String>>();
+                            let mut vals = pending_values.iter().cloned().collect::<Vec<String>>();
```

Comment:
Can't this code just live inside the above if statement? `if !pending_values.is_empty() && allow_propose`?

Reply:
yes, you are correct. this logic can be condensed a few levels.